packages/ast-spec/src/base/FunctionBase.ts

```diff
@@ -0,0 +1,74 @@
+import type { Identifier } from '../expression/Identifier/spec';
+import type { TSTypeAnnotation } from '../special/TSTypeAnnotation/spec';
+import type { TSTypeParameterDeclaration } from '../special/TSTypeParameterDeclaration/spec';
+import type { BlockStatement } from '../statement/BlockStatement/spec';
+import type { Expression } from '../unions/Expression';
+import type { Parameter } from '../unions/Parameter';
+import type { BaseNode } from './BaseNode';
+
+export interface FunctionBase extends BaseNode {
+  /**
+   * True if the function is async:
+   * ```
+   * async function foo(...) {...}
+   * const x = async function (...) {...}
+   * const x = async (...) => {...}
+   * ```
+   */
+  async: boolean;
+  /**
+   * The body of the function.
+   * - For an `ArrowFunctionExpression` this may be an `Expression` or `BlockStatement`.
+   * - For a `FunctionDeclaration` or `FunctionExpression` this is always a `BlockStatement.
+   * - For a `TSDeclareFunction` this is always `undefined`.
+   * - For a `TSEmptyBodyFunctionExpression` this is always `null`.
+   */
+  body?: BlockStatement | Expression | null;
+  /**
+   * This is only true if and only if the node is a `TSDeclareFunction` and it has `declare:
```

Comment:
Slightly more confident about this phrasing suggestion 😄 

```suggestion
   * Whether the node is a `TSDeclareFunction` and it has `declare:
```